core/services/pipeline/task.estimategas.go

```diff
@@ -69,6 +72,7 @@ func (t *EstimateGasLimitTask) Run(ctx context.Context, lggr logger.Logger, vars
 		// Default to 1, i.e. exactly what estimateGas suggests
 		errors.Wrap(ResolveParam(&multiplier, From(VarExpr(t.Multiplier, vars), NonemptyString(t.Multiplier), decimal.New(1, 0))), "multiplier"),
 		errors.Wrap(ResolveParam(&chainID, From(VarExpr(t.getEvmChainID(), vars), NonemptyString(t.getEvmChainID()), "")), "evmChainID"),
+		errors.Wrap(ResolveParam(&block, From(VarExpr(t.Block, vars), t.Block)), "block"),
```

Comment:
Do we risk an "ErrParameterEmpty" if block is not provided here?

Reply:
`VarExpr(t.Block, vars)` returns `ErrParameterEmpty ` and this is handled in `From(VarExpr(t.Block, vars), t.Block))`, which returns "" (empty string). 

Then, `selectBlock()` handles empty string by defaulting to latest